docs/src/ci.md

```diff
@@ -0,0 +1,167 @@
+
+## Stage: Generate integration Tests
+
+The integration tests checks, if all (sub-) packages which use the package `orig` as dependency still works with the code modification of the current pull request.
+
+Before we talk about the details, here a small overview what the `integTestGen.jl` script is doing. First the CI needs to determine, which (sub-) packages has `orig` as dependency. Therefore the CI downloads the `integTestGen.jl` script from the `QED.jl` project via `git clone`. The `integTestGen.jl` script finds out, which (sub-) package use `orig` and generates for each (sub-) package a new CI test job. So the output of the `integTestGen.jl` is a GitLab CI yaml file, which can be executed via [GitLab CI child pipeline](https://about.gitlab.com/blog/2020/04/24/parent-child-pipelines/#dynamically-generating-pipelines).
+
+The `integTestGen.jl` script traverses the dependency tree of `QED.jl` package. Because each QED sub-package is a dependency of the `QED.jl` package, the `QED.jl` dependency tree contains implicit all dependency trees of the sub-packages. So the script is traversing the tree and create a list of sub-packages who depends on `orig`. For better understanding we name the (sub-) packages of the list `users`.
+
+For each `user` from the list of `users` we need to define a separate CI job. First the job checkouts the `dev` branch of the git repository of `user`. Then it set the modified version of `orig` as dependency (`Pkg.develop(path="$CI_package_DIR")`). Finally it execute the unit tests of `user`. The unit tests of `user` are tested with the code changes of the current pull request.
+
+If the `dev` branch of `user` is not working, it is also possible to define a custom git commit, which is working via the git commit message of the pull request of `orig`. For more details see [Integration Tests for CI Users](#Integration-Tests-for-CI-Users).
+
+!!! note
+
+    If a sub-package triggers an integration test, the main package `QED.jl` is passive. It does not get any notification or trigger any script. The repository is simply cloned.
+
+!!! note
+
+    The commit message is defined in the environment variable `CI_COMMIT_MESSAGE` by GitLab CI. If the variable is not defined, the script ignores the commit message. If you want to disable reading the commit message, you can set the name of the commit message variable to an undefined variable via the first argument of the `integTestGen.jl` script. We use this when executing the CI on the `main` or `dev` branch. On these branches, it should not be possible to use custom URLs for unit or integration tests. Therefore we disable it, which also allows the use of `CI_INTG_PKG_URL_` variables as regular part of the merge commit message.
```

Comment:
Yes, it can. Normally the renderer should do it but I see GitHub does not do it. The `Documenter.jl` do line breaks:

![image](https://github.com/QEDjl-project/QED.jl/assets/10682583/d5e774bc-12ec-48eb-b6c3-8210bd126a4a)

Also vsc, which is a little bit strange, because I thought vsc and GitHub uses the same renderer:

![image](https://github.com/QEDjl-project/QED.jl/assets/10682583/9b33aedf-0d67-47de-ae68-71209a2d4644)

Nevertheless, I add some line breaks.